iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -249,6 +250,31 @@ public IfExpression ifExpression(SyntaxToken keyword, ParenthesizedExpression co
     return new IfExpressionImpl(keyword, condition, object);
   }
 
+  public Expression memberExpression(Expression value, List<MemberExpression> memberExpressionComponent) {
+    Expression result = value;
+    for (MemberExpression memberExpression : memberExpressionComponent) {
+      result = ((MemberExpressionImpl) memberExpression).complete(result);
+    }
+    return result;
+  }
+
+  public MemberExpression memberExpressionComponent(SyntaxToken firstToken, Identifier identifier) {
+    StringLiteralImpl identifierAsStringLiteral = new StringLiteralImpl(((IdentifierImpl) identifier).getToken());
+    return new MemberExpressionImpl(firstToken, identifierAsStringLiteral);
+  }
+
+  public MemberExpression memberExpressionComponent(SyntaxToken firstToken, Expression expression, SyntaxToken secondToken) {
```

Comment:
IMO the tokens should be named `openBracket` and `closeBracke` or something similar. Right now it is hard to see from which syntax the method is called.